azure-keyvault/azure/keyvault/custom/key_vault_client.py

```diff
@@ -0,0 +1,75 @@
+#---------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+#---------------------------------------------------------------------------------------------
+
+import uuid
+from msrest.pipeline import ClientRawResponse
+
+from ..key_vault_client import KeyVaultClient as KeyVaultClientBase
+from ..models import KeyVaultErrorException
+
+
+class CustomKeyVaultClient(KeyVaultClientBase):
+
+    def __init__(self, credentials):
+        super(CustomKeyVaultClient, self).__init__(credentials)
+
+    def get_pending_certificate_signing_request(self, vault_base_url, certificate_name, custom_headers=None, raw=False, **operation_config):
+        """Gets the Base64 pending certificate signing request (PKCS-10).
+
+        :param vault_base_url: The vault name, e.g.
+         https://myvault.vault.azure.net
+        :type vault_base_url: str
+        :param certificate_name: The name of the certificate
+        :type certificate_name: str
+        :param dict custom_headers: headers that will be added to the request
+        :param bool raw: returns the direct response alongside the
+         deserialized response
+        :param operation_config: :ref:`Operation configuration
+         overrides<msrest:optionsforoperations>`.
+        :return: Base64 encoded pending certificate signing request (PKCS-10).
+        :rtype: str
+        :rtype: :class:`ClientRawResponse<msrest.pipeline.ClientRawResponse>`
+         if raw=true
+        """
+        # Construct URL
+        url = '/certificates/{certificate-name}/pending'
+        path_format_arguments = {
```

Comment:
relocated plus needed to change self.keyvault.* to self.* since the class is now a subclass rather than wrapping the generated KeyVaultClient

---
In reply to: [114899970](https://github.com/Azure/azure-sdk-for-python/pull/1137#discussion_r114899970) [](ancestors = 114899970)